.github/workflows/maturin.yml

```diff
@@ -0,0 +1,42 @@
+# This file is was initially generated by maturin generate-ci github, then heavily simplified.
+
+name: Maturin
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - '*' # TODO: which tags matter?
```

Comment:
How do we plan to tag releases? So that we only generate the wheels artifacts on release tags.

Reply:
this is the pattern we have been using for python and js: `python-v0.5.0`, so something like `rust-v*` should work?